src/lib/InputsResolver.ts

```diff
@@ -0,0 +1,39 @@
+import {Inputs} from "../types";
+import normalize = require('normalize-path');
+import path = require('path');
+import untildify = require("untildify");
+
+/**
+ * Service for processing inputs (like "--path", "--target", "--projectdir", etc) to get resolved/normalized outputs
+ */
+export interface InputsResolver {
```

Comment:
I don't particularly like the name of this service... because currently it just works on resolving path based stuff. I originally had it called PathResolver but we resolve paths in other places independent of the inputs (i.e. the parsed flags as we used to call it). Leaving this name here for now... but i'm sure with future refactorings  - we'll see what organically evolves.

Reply:
I continued refactoring with https://github.com/forcedotcom/sfdx-scanner/pull/1304 and so this name will ultimately end up being InputProcessor when all said and done.